src/StreamFetcher.ts

```diff
@@ -11,111 +11,126 @@ const logger = getLogger('streamr:StreamFetcher')
 const MAX_AGE = 15 * 60 * 1000 // 15 minutes
 const MAX_AGE_MINUTE = 1000 // 1 minutes
 
-function formHeaders(sessionToken: string|undefined) {
-    const headers: any = {}
+function formHeaders(sessionToken?: string) {
+    const headers: Headers = new Headers()
     if (sessionToken) {
-        headers.Authorization = `Bearer ${sessionToken}`
+        headers.set('Authorization', `Bearer ${sessionToken}`)
     }
     return headers
 }
 
-async function fetchWithErrorLogging(...args: Todo[]) {
+async function fetchWithErrorLogging(...args: Parameters<typeof fetch>) {
     try {
-        // @ts-expect-error
         return await fetch(...args)
-    } catch (e) {
-        logger.error(`failed to communicate with E&E: ${e}`)
-        throw e
+    } catch (err) {
+        logger.error(`failed to communicate with core-api: ${err}`)
+        throw err
     }
 }
 
-async function handleNon2xxResponse(funcName: Todo, response: Todo, streamId: string, sessionToken: string|undefined, method: Todo, url: string) {
-    const errorMsg = await response.text()
+async function handleNon2xxResponse(
+    funcName: string,
+    response: Response,
+    streamId: string,
+    sessionToken: string|undefined|null,
+    method: string,
+    url: string,
+): Promise<HttpError> {
+    const errorMsg = await response.text().catch((err) => {
+        logger.warn('Error reading response text: %s', err)
+        return ''
+    })
     logger.debug(
-        '%s failed with status %d for streamId %s, sessionToken %s : %o',
+        '%s failed with status %d for streamId %s, sessionToken %s: %o',
         funcName, response.status, streamId, sessionToken, errorMsg,
     )
-    throw new HttpError(response.status, method, url)
+    return new HttpError(response.status, method, url)
```

Comment:
This now returns an error rather than throwing, mainly to make typescript happy. 
If you `return handleNon2xxResponse(…)` inside another function, TypeScript will complain because the return type of `handleNon2xxResponse` doesn't match the calling function's return type, even though `handleNon2xxResponse` always throws.